source/common/config/eds_grpc_subscription_impl.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include "envoy/api/v2/core/base.pb.h"
+#include "envoy/config/subscription.h"
+#include "envoy/event/dispatcher.h"
+#include "envoy/grpc/async_client.h"
+
+#include "common/config/grpc_mux_impl.h"
+#include "common/config/grpc_mux_subscription_impl.h"
+#include "common/config/utility.h"
+
+namespace Envoy {
+namespace Config {
+
+class EdsGrpcSubscriptionImpl : public Config::Subscription<envoy::api::v2::ClusterLoadAssignment> {
```

Comment:
I'm still a bit confused what this is providing over `GrpcSubscriptionImpl`? They seem to have the same implementation.

Reply:
Meant to ask you about this and then forgot. My understanding is that the instance of ```GrpcMux``` in ```GrpcSubscriptionImpl``` will be deallocated together with the main class. I need to pass a reference to ```GrpcMux``` to ```EdsGrpcSubscriptionImpl```, as the mux instances are managed by ```EdsSubscriptionFactory```. Subscription classes are very much the same, but I wasn't sure how to reconcile mux ownership...

Reply:
Sure, I think we can refactor this a bit, have a base `GrpcSubscriptionImpl` that takes a `GrpcMux&` and then an `OwnedGrpcSubscriptionImpl` subclass that creates and owns the `GrpcMux` for those use cases.